transcom/mymove
# INT B-18313 Multi-Moves Profile Validation & Create a Move Btn Functionality INTEGRATION (PUSH AFTER B-18312)

INTEGRATION

## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A873918&RoomContext=TeamRoom%3A808731&concept=TeamRoom)

## Summary

As we transition to allowing for multiple moves, we need to allow for the customer to validate their profile prior to starting a new move. If the service member does not have a previous move, then there's no need for them to validate their profile - technically it should already be current.

I went ahead and leveraged the `useLocation` hook to pass around a state value (`needsToVerifyProfile`) when a user is validating their profile. At first I was using `searchParams` but it felt a little clunky and was a lot harder to test, so I transitioned to using local `state` and passing that `state` to the relevant components.

This PR does the following:
- updates profile links to pass around the `state` object holding the current value of `needsToVerifyProfile`
- based on that state, specifically in the `Profile` component, conditional rendering will occur of links and an alert banner
- updated tests

### How to test Scenario 1 (previous moves)

1. Access MM as a customer
2. Once logged in you should be directed to the multi move home page
3. You'll see the multi move dashboard (currently using dummy/static data)
4. With default data, click on the `Create a Move` button
5. This will navigate the user to the profile page with an alert, a `Create a Move` button, and `Return to Dashboard` (instead of `Return to Move`)
6. Validate any data/cancel any data and the alert/button/dashboard link should remain
7. Once you're finished, click the `Create a Move` button and you'll be navigated to the current home page

### How to test Scenario 2 (no previous moves)

1. Access MM as a customer
2. Once logged in, put in `http://milmovelocal:3000/?moveData=noMoves` into the url
3. You'll see the multi move dashboard with no current or previous moves
4. Click on the `Create a Move` button
5. This will navigate the user to the current home page since there is no previous data to validate

## Screenshots

![Screenshot 2024-01-25 at 2 19 15 PM](https://github.com/transcom/mymove/assets/136510600/ddca29ff-6a6c-4af6-a008-2814c35e8756)
